crates/accelerate/src/two_qubit_decompose.rs

```diff
@@ -180,20 +272,919 @@ fn __num_basis_gates(basis_b: f64, basis_fidelity: f64, unitary: MatRef<c64>) ->
+                requested_fidelity: fidelity,
+                calculated_fidelity: -1.0,
+                unitary_matrix,
+            },
+        };
+
+        let tr = if flipped_from_original {
+            let [da, db, dc] = [
+                PI2 - a - specialized.a,
+                b - specialized.b,
+                -c - specialized.c,
+            ];
+            4. * Complex64::new(
+                da.cos() * db.cos() * dc.cos(),
+                da.sin() * db.sin() * dc.sin(),
+            )
+        } else {
+            let [da, db, dc] = [a - specialized.a, b - specialized.b, c - specialized.c];
+            4. * Complex64::new(
+                da.cos() * db.cos() * dc.cos(),
+                da.sin() * db.sin() * dc.sin(),
+            )
+        };
+        specialized.calculated_fidelity = trace_to_fid(tr);
+        if let Some(fid) = specialized.requested_fidelity {
+            if specialized.calculated_fidelity + 1.0e-13 < fid {
+                import_exception!(qiskit, QiskitError);
+                return Err(QiskitError::new_err(format!(
+                    "Specialization: {:?} calculated fidelity: {} is worse than requested fidelity: {}",
+                    specialized.specialization,
+                    specialized.calculated_fidelity,
+                    fid
+                )));
+            }
+        }
+        specialized.global_phase += tr.arg();
+        Ok(specialized)
+    }
+
+    #[allow(non_snake_case)]
+    #[getter]
+    fn K1l(&self, py: Python) -> PyObject {
+        self.K1l.to_pyarray(py).into()
+    }
+
+    #[allow(non_snake_case)]
+    #[getter]
+    fn K1r(&self, py: Python) -> PyObject {
+        self.K1r.to_pyarray(py).into()
+    }
+
+    #[allow(non_snake_case)]
+    #[getter]
+    fn K2l(&self, py: Python) -> PyObject {
+        self.K2l.to_pyarray(py).into()
+    }
+
+    #[allow(non_snake_case)]
+    #[getter]
+    fn K2r(&self, py: Python) -> PyObject {
+        self.K2r.to_pyarray(py).into()
+    }
+
+    #[getter]
+    fn unitary_matrix(&self, py: Python) -> PyObject {
+        self.unitary_matrix.to_pyarray(py).into()
+    }
+
+    #[pyo3(signature = (euler_basis=None, simplify=false, atol=None))]
+    fn circuit(
+        &mut self,
+        euler_basis: Option<&str>,
+        simplify: bool,
+        atol: Option<f64>,
+    ) -> TwoQubitGateSequence {
+        let binding = self.default_euler_basis.clone();
+        let euler_basis: &str = euler_basis.unwrap_or(&binding);
+        let target_1q_basis_list: Vec<&str> = vec![euler_basis];
+
+        let mut gate_sequence = Vec::new();
+        let mut global_phase: f64 = self.global_phase;
+
+        let c2r = unitary_to_gate_sequence_inner(
+            self.K2r.view(),
+            &target_1q_basis_list,
+            0,
+            None,
+            simplify,
+            atol,
+        )
+        .unwrap();
+        for gate in c2r.gates {
+            gate_sequence.push((gate.0, gate.1, [0, 0]))
+        }
```

Comment:
I just switched it up to smallvec and did this in a single go. I only did this to avoid the extra vec overhead because it's only 1 or 2 elements, I forgot about smallvec for some reason. See: https://github.com/Qiskit/qiskit/pull/11946/commits/aff637eb0d830b26d16c978a50e86d52eb06bf2c